<launch>
    <arg name="cheetah_namespace" default="cheetah" />
    <param name="use_sim_time" value="false" />


    <!-- if debug -->

    <!-- <node name="cheetah_inekf_lcm" pkg="cheetah_inekf_lcm" type="cheetah_estimator" output="screen" required="true" launch-prefix="gdb -ex run args"/> -->
    <node name="cheetah_inekf_lcm" pkg="cheetah_inekf_lcm" type="cheetah_estimator" output="screen" required="true"/>


    <!-- Run cheetah estimator publisher -->
    <rosparam file="$(find cheetah_inekf_lcm)/config/settings.yaml" command="load" />
    <!-- Initialize inEKF params -->
    <rosparam file="$(find cheetah_inekf_lcm)/config/inekf.yaml" command="load"/>
    <!-- Initialize inEKF noise -->
    <rosparam file="$(find cheetah_inekf_lcm)/config/noise.yaml" command="load"/>
    <!-- Initialize inEKF prior -->
    <rosparam file="$(find cheetah_inekf_lcm)/config/prior.yaml" command="load"/>

    
    <node name="InEKF_path_publisher" pkg="cheetah_inekf_lcm" type="path_publisher_cheetah_node" output="screen">
      <param name="pose_topic" value="/cheetah/inekf_estimation/pose"/>
      <param name="path_topic" value="/cheetah/inekf_estimation/path"/>
      <param name="publish_rate" value="2"/>
      <param name="pose_skip" value="100"/>
      <param name="use_sim_time" value="false" />
      <!-- <param name="file_name" value="/media/curly_ssd_justin/code/minicheetah-perception/catkin_ws/src/cheetah_inekf_lcm/data/08292020_trail1_gt_2.txt"/> -->
    </node>
</launch>